File: examples/android/app/src/test/java/org/aetherfeed/app/news/ArticleIndexTest.kt
package org.aetherfeed.app.news

import org.aetherfeed.app.domain.Article
import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import org.robolectric.annotation.Config

@RunWith(RobolectricTestRunner::class)
@Config(sdk = [26])
class ArticleIndexTest {
    @Test
    fun roundTripsHeadlineRows() {
        val rows = listOf(
            Article(
                id = "feed-1:1",
                feedId = "feed-1",
                title = "Hello",
                url = "https://example.invalid/1",
                publishedAt = 10L,
                summary = "Sum",
                contentHtml = "<p>Body</p>",
            ),
        )
        val decoded = decodeArticleIndex(encodeArticleIndex(rows))
        assertEquals(rows, decoded)
    }

    @Test
    fun emptyAndJunkDecodeToNone() {
        assertTrue(decodeArticleIndex("").isEmpty())
        assertTrue(decodeArticleIndex("not-json").isEmpty())
        assertTrue(decodeArticleIndex("[{}]").isEmpty())
    }
}

File: examples/web/src/news/articleIndex.test.ts
import { describe, expect, it } from "vitest";
import { parseArticleIndex } from "./articleIndex";

describe("article index", () => {
  it("keeps valid headline rows and drops junk", () => {
    const rows = parseArticleIndex([
      {
        id: "feed-1:1",
        feedId: "feed-1",
        title: "Hello",
        url: "https://example.invalid/1",
        publishedAt: 10,
        summary: "Sum",
      },
      { id: "", feedId: "feed-1", title: "Bad", url: "https://example.invalid/x" },
      "nope",
    ]);
    expect(rows).toEqual([
      {
        id: "feed-1:1",
        feedId: "feed-1",
        title: "Hello",
        url: "https://example.invalid/1",
        publishedAt: 10,
        summary: "Sum",
        contentHtml: undefined,
      },
    ]);
    expect(parseArticleIndex(null)).toEqual([]);
  });
});
